### PR TITLE
Add support for docker build caching, disabled by default

### PR DESCRIPTION
**Pull request by ryanwi**
Allow projects to use docker build cache

### CODE REVIEW

File: .github/actions/docker/action.yml [hunk 1]
```diff
@@ -63,13 +63,9 @@ inputs:
     default: 'false'
     description: 'Generate tag only.'
     required: false
-  CACHE_FROM:
```
Comment (ryanwi) [inline .github/actions/docker/action.yml:66]: these are not used by any consuming repos and not exposed in the ci workflows so removing them

File: .github/actions/docker/action.yml [hunk 2]
```diff
@@ -126,15 +122,31 @@ runs:
       run: echo BUILD_ARGS=${{inputs.BUILD_ARGS}} >> $GITHUB_ENV
       shell: bash
 
-    - name: Build and export to Docker
+    - name: Build and export to Docker without cache
       uses: docker/build-push-action@v6
-      if: inputs.TAG_ONLY == 'false'
```
Comment (ryanwi) [inline .github/actions/docker/action.yml:131]: TAG_ONLY is never used as far as I can tell, in any consuming repos, so removing

Comment (Null-Is-Null) [reply .github/actions/docker/action.yml:131]: `TAG_ONLY` looks to be used in `workflows/cd-retag.yml`. We should remove references to `TAG_ONLY` there as well.

File: .github/actions/docker/action.yml [hunk 3]
```diff
@@ -63,13 +63,9 @@ inputs:
     default: 'false'
     description: 'Generate tag only.'
     required: false
```
Comment (Null-Is-Null) [inline .github/actions/docker/action.yml:65]: This input should be removed as well if we're eliminating `TAG_ONLY` entirely.

**Review by Null-Is-Null — CHANGES_REQUESTED** [review]

File: .github/actions/docker/action.yml (hunk 3 above)
Comment (ryanwi) [reply .github/actions/docker/action.yml:65]: i can keep it if we think it is or should be used? would need to understand the use case that led to it being added

Comment (Null-Is-Null) [reply .github/actions/docker/action.yml:65]: From what I can see, the only use I've seen so far is in the cd-retag workflow, which tags an existing image with another tag and pushes it. Let's keep things as-is for now.

**Review by Null-Is-Null — APPROVED** [review]

**Pull request merged**